skopt/optimizer/optimizer.py

```diff
@@ -467,7 +471,7 @@ def tell(self, x, y, fit=True):
                 # minimization starts from `n_restarts_optimizer` different
                 # points and the best minimum is used
                 elif self.acq_optimizer == "lbfgs":
-                    x0 = X[np.argsort(values)[:self.n_restarts_optimizer]]
+                    x0 = X[np.argsort(values)]
```

Comment:
I read the two comments (https://github.com/scikit-optimize/scikit-optimize/pull/284#issuecomment-270078807 and following) as: "running BFGS after sampling doesn't really improve things much over purely sampling" and "BFGS with only `n_restarts_optimizer` random points isn't very good at all"

@iaroslav-ai yes for trees we can't use BFGS, so all we do is sample to try and find the optimum. I was wondering for the case where you can use BFGS if we would do better by distribution our budget of 10000 evaluations of the acquisition function differently between random sampling, different starting points, and iterations of each BFGS run